### PR TITLE
add listener for history.pushState to detect SPA navigation

### PR DESCRIPTION
**Pull request by lucasfischer**
At the moment Fathom can't tell if the there was a url change in a SPA, because it only creates a tracking image when a page was loaded. According to my understanding this is done via `trackPageview()`. In this pull request i propose listening on the `window.history.pushState` event to detect if an url has changed and fire `trackPageview()` again. This would have the advantage, that react, vue, angular, etc. applications can be tracked correctly instead of just tracking the initial page load.

I really would like to see this feature in the future. If something is wrong with my PR (it's my first one ever), please let me know.

### CODE REVIEW

File: assets/src/js/tracker.js [hunk 1]
```diff
@@ -171,6 +171,23 @@ function trackPageview() {
   window.setTimeout(() => { document.body.removeChild(i)}, 1000);
 }
 
+// add listener for history.pushState to detect SPA navigation
+var his = window.history
```
Comment (antsmartian) [inline assets/src/js/tracker.js]: `his`?

Comment (antsmartian) [reply assets/src/js/tracker.js]: Also I'm seeing arrow functions used in this file, which means you can use them and get rid of handling `this` yourselves.

Comment (lisovskyvlad) [reply assets/src/js/tracker.js]: better to name it "she" 🥇

Comment (lucasfischer) [reply assets/src/js/tracker.js]: i renamed it to `h`

**Comment by lucasfischer** [comment]
Hey @dannyvankooten,

Is there anything to add, before this PR can be merged?

I really need this feature to track my services correctly, and I think others will too.

**Comment by lucasfischer** [comment]
Thanks for your feedback and the tip with `fathom('trackPageview')`. Maybe you can add this to the wiki?

These are all reasonable points. The problem is that there is no other way to tell when history.pushState was triggered.

what do you think about making this an opt-in feature?
For example: `fathom('detectHistoryPushState', true)`

**Comment by JackEllis** [comment]
@lucasfischer Really appreciate the PR but this is too opinionated to bring in. This should be handled manually by those building SPAs

**Pull request closed without merging**